### ISSUE TITLE
Traffic filters in 0.6.0: "provider produced inconsistent result after apply"

### ISSUE THREAD

**Issue by jamieparkinson** [elastic/terraform-provider-ec#621]
## Readiness Checklist
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
Planning of an `ec_deployment` with associated `ec_deployment_traffic_filter_association`s should be idempotent: after applying once no changes should remain in the plan, and subsequent applys should succeed regardless.

## Current Behavior
A cluster with associated traffic filters has successfully been configured/applied, but a subsequent plan aims to remove the traffic filters. On applying this plan, the following error occurs (specifics have been replaced with pseudocode in angle brackets):

```
Error: Provider produced inconsistent result after apply

When applying changes to <ec_deployment resource>, provider "provider[\"registry.terraform.io/elastic/ec\"]" produced an unexpected new value: .traffic_filter: was null, but now cty.SetVal([]cty.Value{cty.StringVal("<filter id 1>"), cty.StringVal("<filter id 2>")}).

This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

We are also seeing spurious config changes to the `elasticsearch` block in the deployment resource, but I think this is already being tracked in other issues on this repo.

## Terraform resources
```hcl
resource "ec_deployment" "cluster" {
  // name, alias etc config omitted for brevity
  elasticsearch = {
    autoscale = "false"
    hot = {
      size        = "1g"
      zone_count  = 1
      autoscaling = {}
    }
  }
}

resource "ec_deployment_traffic_filter_association" "cluster" {
  for_each = var.network_config.ec_traffic_filters // set of filter ids

  deployment_id     = ec_deployment.cluster.id
  traffic_filter_id = each.value
}
```

## Steps to Reproduce

1. Apply a plan with `ec_deployment_traffic_filter_association`s
2. Plan again - see that filters are to be removed
3. Apply again - see error above.

## Your Environment
Using the 0.6.0 provider

**Comment by jamieparkinson** [comment]
Related to https://github.com/elastic/terraform-provider-ec/issues/419

**Comment by tobio** [comment]
This got released with 0.8.0 yesterday.
